multus_sriov_cni_install.sh

```diff
@@ -0,0 +1,202 @@
+    "kubeconfig": "$KUBECONFIG"
+}
+EOF
+    cp $CNI_CONF_DIR/00-multus.conf $ARTIFACTS
+    return $?
+}
+
+
+function download_cni {
+    if ! $RECLONE ; then
+        return 0
+    fi
+    #trap read debug
+
+    echo "Download $MULTUS_CNI_REPO"
+    git clone $MULTUS_CNI_REPO $WORKSPACE/multus-cni
+    cd $WORKSPACE/multus-cni
+    git checkout $MULTUS_CNI_BRANCH
+    git log -p -1 > $ARTIFACTS/multus-cni.txt
+    cd -
+
+    echo "Download $SRIOV_CNI_REPO"
+    git clone $SRIOV_CNI_REPO $WORKSPACE/sriov-cni
+    pushd $WORKSPACE/sriov-cni
+    git checkout $SRIOV_CNI_BRANCH
+    git log -p -1 > $ARTIFACTS/sriov-cni.txt
+    popd
+
+    echo "Download $PLUGINS_REPO"
+    git clone $PLUGINS_REPO $WORKSPACE/plugins
+    pushd $WORKSPACE/plugins
+    git checkout $PLUGINS_BRANCH
+    git log -p -1 > $ARTIFACTS/plugins.txt
+    bash ./build_linux.sh
+    cp $WORKSPACE/plugins/bin/* $CNI_BIN_DIR/
+    popd
+
+    echo "Download $SRIOV_NETWORK_DEVICE_PLUGIN_REPO"
+    git clone $SRIOV_NETWORK_DEVICE_PLUGIN_REPO $WORKSPACE/sriov-network-device-plugin
+    pushd $WORKSPACE/sriov-network-device-plugin
+    git checkout $SRIOV_NETWORK_DEVICE_PLUGIN_BRANCH
+    git log -p -1 > $ARTIFACTS/sriov-network-device-plugin.txt
+    make build
+    cp $WORKSPACE/sriov-network-device-plugin/build/sriovdp $CNI_BIN_DIR
+
+    #TODO still missing sriov in $CNI_BI_DIR
+    popd
+    mkdir -p /etc/pcidp/
+    cat > /etc/pcidp/config.json <<EOF
+{
+    "resourceList": [{
+        "resourceName": "sriov",
+        "selectors": {
+                "vendors": ["15b3"],
+                "devices": ["1018"],
+                "drivers": ["mlx5_core"]
+            }
+    }
+    ]
+}
+EOF
+    cp /etc/pcidp/config.json $ARTIFACTS
+    return 0
+}
+
+
+function create_vfs {
+    echo 0 > /sys/class/net/$SRIOV_INTERFACE/device/sriov_numvfs
+    echo $VFS_NUM > /sys/class/net/$SRIOV_INTERFACE/device/sriov_numvfs
+}
+
+
+function install_k8s {
+    echo "Stopping previous K8S run"
+    #TODO clean on exit
+    kill $(ps -ef |grep kube |awk '{print $2}')
+    go get -d k8s.io/kubernetes
+    cd $GOPATH/src/k8s.io/kubernetes
+    git log -p -1 > $ARTIFACTS/kubernetes.txt
+    make
+    go get -u github.com/tools/godep
+    go get -u github.com/cloudflare/cfssl/cmd/...
+    kill -9 $(ps -ef |grep etcd|grep http|awk '{print $2}')
+    $GOPATH/src/k8s.io/kubernetes/hack/install-etcd.sh
+    $GOPATH/src/k8s.io/kubernetes/hack/local-up-cluster.sh 2>&1|tee > $LOGDIR/kubernetes.log &
+    kubectl get pods
+    rc=$?
+    #TODO add timeout and better k8s up check
+    while [ $rc -ne 0 ]; do
+       echo "Wait until K8S is up"
+       kubectl get pods
+       rc=$?
+       sleep 5
+    done
+    return $k8s_pid
+}
+
+
+#TODO add check prereqs
+#TODO add docker image mellanox/mlnx_ofed_linux-4.4-1.0.0.0-centos7.4 presence
+
+create_vfs
+download_cni
+
+install_k8s
+configure_multus
+while [ ! -f $WORKSPACE/sriov-network-device-plugin/deployments/crdnetwork.yaml ]; do
```

Comment:
This file will be exist once `sriov-network-device-plugin` repo is cloned.
Instead of checking the file existence, we shall create the crd by running `kubectl create -f $WORKSPACE/sriov-network-device-plugin/deployments/crdnetwork.yaml`